### PR TITLE
[3.x] Add C# code examples (salvage of VinnNo's PRs)

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR is mostly salvages of @VinnNo's PRs that add C# code examples, with lots of changes. It also updates the GDScript docs in a few places, targeting Godot 3.x.

### CODE REVIEW

**Comment by mhilbrunner** [comment]
Thanks so much for working on salvaging these, I'd be a shame for them to be left unused :)
I'd be grateful if @paulloz can review this

File: tutorials/math/random_number_generation.rst [hunk 1]
```diff
@@ -87,83 +127,169 @@ other things.
...
+        for (int i = 0; i < 100; i++)
+        {
+            // Pick 100 fruits randomly.
+            GD.Print(GetFruit());
+        }
+    }
+
+    public string GetFruit()
+    {
+        var randomFruit = _fruits[GD.Randi() % _fruits.Length];
```
[the remark this answers is not in the document]
Comment (aaronfranke) [reply tutorials/math/random_number_generation.rst]: Consistency with GDScript is more important here.

**Review by paulloz — APPROVED** [review]

File: tutorials/math/beziers_and_curves.rst [hunk 2]
```diff
@@ -164,6 +215,15 @@ Let's do a simple example with the following pseudocode:
...
         position = _cubic_bezier(p0, p1, p2, p3, t)
 
+ .. code-tab:: csharp
+
+    private float _t = 0.0f;
+
+    public override void _Process(float delta)
+    {
+        _t += delta;
+        Position = CubicBezier(p0, p1, p2, p3, _t)
```
Comment (hpnrep6) [inline tutorials/math/beziers_and_curves.rst:225]:
```suggestion
        Position = CubicBezier(p0, p1, p2, p3, _t);
```

**Review by hpnrep6 — COMMENTED** [review]
Missing semicolon

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thank you, and thanks to the reviewers! :)

**Comment by VinnNo** [comment]
Hey! I didn't even know these were getting attention. So, I'm still VERY interested in proceeding with expanding the C# documentation, but honestly, I'm not sure how to structure these. I find that the nested code tabs conflict with the document bodies often, where variable names are mentioned (EG some_string vs SomeString) as GD and C# have two different style-guides. Also what should the approach for pages like [this](https://docs.godotengine.org/en/stable/classes/class_animationnode.html) be? Anyway, thanks for salvaging these and improving them!

**Comment by VinnNo** [comment]
I almost want to suggest a doc arrangement like [this](https://cucumber.io/docs/guides/api-automation/) for the two languages, but I understand that is a big undertaking but it does read very well.

**Pull request merged**